src/main/java/com/gojek/beast/sink/bq/handler/impl/OOBErrorHandler.java

```diff
@@ -47,13 +54,12 @@ public Status handleErrorRecords(Map<Record, List<BQInsertionRecordsErrorType>>
                 //add the record into write list
                 recordsToWrite.add(record);
             }
-            if (recordHasInvalidData) {
-                shouldBatchFail = true;
-            }
         }
         if (shouldBatchFail) {
```

Comment:
the metric for OOB message count is captured in GCSWriter, that will be only sent if we enable ErrorHandler as GcsWriter.
It is better to send metric about OOB message count from here as well, since we can use DefaultHandler instead of GCSWriter which just prints the messages.

Reply:
Makes sense. I'll add the OOB count to DefaultWriter.